data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSourceBufferAccumulator.java

```diff
@@ -0,0 +1,151 @@
+
+import org.opensearch.dataprepper.plugins.kafka.util.MessageFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * * A helper utility class which helps to write different formats of records
+ * like json, avro and plaintext to the buffer.
+ */
+@SuppressWarnings("deprecation")
+public class KafkaSourceBufferAccumulator<K, V> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaSourceBufferAccumulator.class);
+    private static final String MESSAGE_KEY = "message";
+    private final TopicsConfig topicConfig;
+    private final KafkaSourceConfig kafkaSourceConfig;
+    private final String schemaType;
+    private PluginMetrics pluginMetrics;
+    private final Counter kafkaConsumerWriteError;
+    private static final String KAFKA_CONSUMER_BUFFER_WRITE_ERROR = "kafkaConsumerBufferWriteError";
+    private static final int MAX_FLUSH_RETRIES_ON_IO_EXCEPTION = Integer.MAX_VALUE;
+    private static final Duration INITIAL_FLUSH_RETRY_DELAY_ON_IO_EXCEPTION = Duration.ofSeconds(5);
+    private final JsonFactory jsonFactory = new JsonFactory();
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    public KafkaSourceBufferAccumulator(final TopicsConfig topicConfigs,
+                                        final KafkaSourceConfig kafkaSourceConfig,
+                                        final String schemaType, PluginMetrics pluginMetric) {
+        this.kafkaSourceConfig = kafkaSourceConfig;
+        this.topicConfig = topicConfigs;
+        this.schemaType = schemaType;
+        this.pluginMetrics = pluginMetric;
+        this.kafkaConsumerWriteError = pluginMetrics.counter(KAFKA_CONSUMER_BUFFER_WRITE_ERROR);
+    }
+
+    public Record<Object> getEventRecord(final String line) {
+        Map<String, Object> message = new HashMap<>();
+        if (MessageFormat.getByMessageFormatByName(schemaType)
+                .equals(MessageFormat.PLAINTEXT)) {
+            message.put(MESSAGE_KEY, line);
+        } else if (MessageFormat.getByMessageFormatByName(schemaType)
+                .equals(MessageFormat.JSON)) {
+            try {
+                final JsonParser jsonParser = jsonFactory.createParser(line);
+                message = objectMapper.readValue(jsonParser, Map.class);
+            } catch (Exception e) {
+                LOG.error("Unable to parse json data [{}], assuming plain text", line, e);
+                message.put(MESSAGE_KEY, line);
+            }
+
+        }
+        Event event = JacksonLog.builder().withData(message).build();
+        return new Record<>(event);
+    }
+
+    public void write(List<Record<Object>> kafkaRecords, final Buffer<Record<Object>> buffer) throws Exception {
+        try {
+            writeAllRecordToBuffer(kafkaRecords,
+                    buffer, topicConfig);
+            LOG.info("Total number of records publish in buffer {} for Topic : {}", kafkaRecords.size(), topicConfig.getName());
```

Comment:
We should remove this log message otherwise there'll be too many logs.

Reply:
Done